config/v1/types_infrastructure.go

```diff
@@ -189,6 +192,50 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+// CloudControllerManagerState defines whether Cloud Controller Manager presence is expected or not
+type CloudControllerManagerState string
+
+const (
+	// Cloud Controller Manager is enabled and expected to be installed.
+	// This value indicates that new nodes should be tainted as uninitialized when created,
+	// preventing them from running workloads until they are initialized by the cloud controller manager.
+	CloudControllerManagerExternal CloudControllerManagerState = "External"
+
+	// Cloud Controller Manager is disabled and not expected to be installed.
+	// This value indicates that new nodes should not be tainted
+	// and no extra node initialization is expected from the cloud controller manager.
+	CloudControllerManagerNone CloudControllerManagerState = "None"
+)
+
+// CloudControllerManagerSpec holds Cloud Controller Manager (a.k.a. CCM or CPI) related settings
+type CloudControllerManagerSpec struct {
+	// state determines whether or not an external Cloud Controller Manager is expected to
+	// be installed within the cluster.
+	// https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+	//
+	// When set to "External", new nodes will be tainted as uninitialized when created,
+	// preventing them from running workloads until they are initialized by the cloud controller manager.
+	// When omitted or set to "None", new nodes will be not tainted
+	// and no extra initialization from the cloud controller manager is expected.
+	// +kubebuilder:validation:Enum="";External;None
+	// +optional
+	State CloudControllerManagerState `json:"state"`
+}
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct {
+	// PlatformName holds the arbitrary string represented cloud provider name, expected to be set at the installation time.
+	// Intended to serve only for informational purposes and not expected to be used for decision-making.
```

Comment:
I don't like using intended here, it makes the statement wooly
```suggestion
	// This field is solely for informational and reporting purposes.
	// The platform does not make decisions based on the value of this field.
```

Reply:
>// The platform does not make decisions based on the value of this field.

I don't feel that we should refer `the platform` here. As I understand, here you meant openshift itself.
Here this word has a different meaning.

Reply:
Refer directly to `OpenShift` instead? Or drop the part about making decisions altogether?

Reply:
In the latest change, i worded it as:
```	
// This field is solely for informational and reporting purposes and is not expected to be used for decision-making.
```
Would this work?